### PR TITLE
Implement authentication middleware

### PR DESCRIPTION
**Pull request by hochiw**
## Features
- Implemented auth context and middleware
- Implemented authed directive
- Implemented sample gql query for getting user address
- Improved gql error handling

based on #138 
refs #119

### CODE REVIEW

File: graphql-server/pkg/context/auth.go [hunk 1]
```diff
@@ -0,0 +1,25 @@
+package context
+
+import (
+	"context"
+)
+
+const (
+	authedUserContextKey contextKey = "authedUserContextKey"
```
Comment (chihimng) [inline graphql-server/pkg/context/auth.go:8]: `AuthedUserContextKey`?

File: graphql-server/pkg/middlewares/auth.go [hunk 2]
```diff
@@ -0,0 +1,49 @@
...
+	"github.com/oursky/likedao/pkg/models"
+)
+
+func Authentication(
+	config config.Config,
+) gin.HandlerFunc {
+	return func(c *gin.Context) {
+
+		if gin.Mode() == gin.DebugMode {
+			dummyAddress := []byte{0, 1, 0, 1, 0, 1, 0, 1, 1, 0, 1, 1, 0}
```
Comment (chihimng) [inline graphql-server/pkg/middlewares/auth.go:20]: Use header to manually override signed in user address?
with this we can debug for unauthed case?

File: graphql-server/pkg/errors/errors.go [hunk 3]
```diff
@@ -1,73 +1,46 @@
...
-
-	w := e.Cause()
-	for {
-		if w == target {
-			return true
-		}
+type ServerErrorCode string
+
+const (
+	InternalError     ServerErrorCode = "INTERNAL_ERROR"
```
Comment (chihimng) [inline graphql-server/pkg/errors/errors.go:13]: should we follow apollo standard codes? https://www.apollographql.com/docs/apollo-server/data/errors/

**Review by chihimng — CHANGES_REQUESTED** [review]

**Comment by hochiw** [comment]
@chihimng updated 🙏

**Review by chihimng — APPROVED** [review]

**Pull request merged**